src/compiler/core.ts

```diff
@@ -784,6 +1054,59 @@ namespace ts {
         return false;
     }
 
+    /**
+     * Extension boundaries by priority. Lower numbers indicate higher priorities, and are
+     * aligned to the offset of the highest priority extension in the
+     * allSupportedExtensions array.
+     */
+    export const enum ExtensionPriority {

```

Comment:
It is a formal way to specify the offsets to each extension in `allSupportedExtensions`, and is used to disambiguate between `file.ts`, and `file.d.ts` (or also `file.js` with `-allowJs`) in the same directory. This could have just been constant values, but I felt this was more descriptive.